peer.go

```diff
@@ -517,6 +522,10 @@ func (p *Peer) connectionCloseStateChange(changed *Connection) {
 
 	if found {
 		p.onClosedConnRemoved(p)
+		// Inform third parties that a peer lost a connection.
+		if p.channel.onPeerStatusChanged != nil {
```

Comment:
My intuition is that, if in the majority of cases, calling the function is avoidable, the nil check is cheaper. However, going forward on the assumption that most use of TChannel will be through YARPC, the nil check is overhead for every change. So, I concluded that using a noopStatusChanged is the way to go.